src/ImageCaptureCore/Defs.cs

```diff
@@ -0,0 +1,292 @@
+		DownloadFailed = -9934,
+		UploadFailed = -9935,
+		FailedToCompletePassThroughCommand = -9936,
+		DownloadCanceled = -9937,
+		FailedToEnableTethering = -9938,
+		FailedToDisableTethering = -9939,
+		FailedToCompleteSendMessageRequest = -9940,
+		DeleteFilesFailed = -9941,
+		DeleteFilesCanceled = -9942,
+		DeviceIsPasscodeLocked = -9943,
+		DeviceFailedToTakePicture = -9944,
+		DeviceSoftwareNotInstalled = -9945,
+		DeviceSoftwareIsBeingInstalled = -9946,
+		DeviceSoftwareInstallationCompleted = -9947,
+		DeviceSoftwareInstallationCanceled = -9948,
+		DeviceSoftwareInstallationFailed = -9949,
+		DeviceSoftwareNotAvailable = -9950,
+		DeviceCouldNotPair = -9951,
+		DeviceCouldNotUnpair = -9952,
+		DeviceNeedsCredentials = -9953,
+		DeviceIsBusyEnumerating = -9954,
+		DeviceCommandGeneralFailure = -9955,
+		DeviceFailedToCompleteTransfer = -9956,
+		DeviceFailedToSendData = -9957,
+		SessionNotOpened = -9958,
+		ThumbnailNotAvailable = ICReturnCodeOffset.Thumbnail,
+		ThumbnailAlreadyFetching = ICReturnCodeOffset.Thumbnail - 1,
+		ThumbnailCanceled = ICReturnCodeOffset.Thumbnail - 2,
+		ThumbnailInvalid = ICReturnCodeOffset.Thumbnail - 3,
+		ErrorDeviceEjected = ICReturnCodeOffset.System,
+		MetadataNotAvailable = ICReturnCodeOffset.Metadata,
+		MetadataAlreadyFetching = ICReturnCodeOffset.Metadata - 1,
+		MetadataCanceled = ICReturnCodeOffset.Metadata - 2,
+		MetadataInvalid = ICReturnCodeOffset.Metadata - 3,
+		MultiErrorDictionary = -30000,
+	}
+	
+	[Native]
+	public enum ICDeviceType : ulong
+	{
+		Camera = 1,
+		Scanner = 2,
+	}
+	
+	[Native]
+	public enum ICDeviceLocationType : ulong
+	{
+		Local = 256,
+		Shared = 512,
+		Bonjour = 1024,
+		Bluetooth = 2048,
+	}
+	
+	[Native]
+	public enum ICDeviceTypeMask : ulong
+	{
+		Camera = 1,
+		Scanner = 2,
+	}
+	
+	[Native]
+	public enum ICDeviceLocationTypeMask : ulong
+	{
+		Local = 256,
+		Shared = 512,
+		Bonjour = 1024,
+		Bluetooth = 2048,
+		Remote = 65024,
+	}
+
+	[Native]
+	public enum ICScannerFunctionalUnitType : ulong
+	{
+		Flatbed = 0,
+		PositiveTransparency = 1,
+		NegativeTransparency = 2,
+		DocumentFeeder = 3,
+	}
+	
+	[Native]
+	public enum ICScannerMeasurementUnit : ulong
+	{
+		Inches = 0,
+		Centimeters = 1,
+		Picas = 2,
+		Points = 3,
+		Twips = 4,
+		Pixels = 5,
+	}
+	
+	[Native]
+	public enum ICScannerBitDepth : ulong
+	{
+		Bits1 = 1,
+		Bits8 = 8,
+		Bits16 = 16,
+	}
+	
+	[Native]
+	public enum ICScannerColorDataFormatType : ulong
+	{
+		Chunky = 0,
+		Planar = 1,
+	}
+	
+	[Native]
+	public enum ICScannerPixelDataType : ulong
+	{
+		Bw = 0,
+		Gray = 1,
+		Rgb = 2,
+		Palette = 3,
+		Cmy = 4,
+		Cmyk = 5,
+		Yuv = 6,
+		Yuvk = 7,
+		Ciexyz = 8,
+	}
+	
+	[Native]
+	public enum ICScannerDocumentType : ulong
+	{
+		Default = 0,
+		A4 = 1,
+		B5 = 2,
+		USLetter = 3,
+		USLegal = 4,
+		A5 = 5,
+		Isob4 = 6,
+		Isob6 = 7,
+		USLedger = 9,
+		USExecutive = 10,
+		A3 = 11,
+		Isob3 = 12,
+		A6 = 13,
+		C4 = 14,
+		C5 = 15,
+		C6 = 16,
+		ICScannerDocumentType4A0 = 17,
```

Comment:
These are likely wrong.

Reply:
Yeah, we have some options here
`_4A0 = 17,`
`Type4A0 = 17,`
`FourA0 = 17,`

Reply:
That should be `4A0`. See the long comment in the header file:

```
@constant ICScannerDocumentType4A0            4A0,                             1682.00 mm x 2378.00 mm
```

Reply:
```csharp> int 4A0 = 42;
(1,5): error CS1525: Unexpected symbol `4'```

Reply:
Yea sorry we can't start with a letter.
Numbers to letters doesn't feel right, for the people who know this stuff it would require a bit of gymnastic if you expect `4A0`.
Then `_` is weird after a `.`: `ICScannerDocumentType._4A0`, it looks internal.
Even if, `Type4A0` would have a redundant `Type` I think it's our best option.

Reply:
You are supposed to use the `DescriptionAttribute` for this kind of scenarios 

```csharp
[Description ("4A0")]
FourA0 = 17,
```

But yeah it is unlikely our tooling does anything with it so my vote is on `Type4A0 = 17,`